setup.py

```diff
@@ -14,32 +14,101 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import re
+import sys
+from operator import lt, gt, eq
+from os.path import (
+    abspath,
+    dirname,
+    join,
+)
+from distutils.version import StrictVersion
 from setuptools import setup, Extension
-from Cython.Build import cythonize
-import numpy as np
+try:
+    from Cython.Build import cythonize
+    from numpy import get_include
+    ext_modules = cythonize(
+        [
+            Extension(
+                'zipline.assets._assets',
+                ['zipline/assets/_assets.pyx'],
+                include_dirs=[get_include()],
+            ),
+            Extension(
+                'zipline.lib.adjusted_array',
+                ['zipline/lib/adjusted_array.pyx'],
+                include_dirs=[get_include()],
+            ),
+            Extension(
+                'zipline.lib.adjustment',
+                ['zipline/lib/adjustment.pyx'],
+                include_dirs=[get_include()],
+            ),
+            Extension(
+                'zipline.data.ffc.loaders._us_equity_pricing',
+                ['zipline/data/ffc/loaders/_us_equity_pricing.pyx'],
+                include_dirs=[get_include()],
+            ),
+        ]
+    )
+except ImportError:
+    if 'build_ext' in sys.argv:
+        raise
+    ext_modules = []
+
+
+def _filter_requirements(lines_iter):
+    for line in lines_iter:
+        line = line.strip()
+        if not line or line.startswith('#'):
+            continue
+
+        # pip install -r understands line with ;python_version<'3.0', but
+        # whatever happens inside extras_requires doesn't.  Parse the line
+        # manually and conditionally add it if needed.
+        if ';' in line:
```

Comment:
The line that this is for is `backports.ssl-match-hostname==3.4.0.2;python_version<'3.0'`.